external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/CachedKafkaProducer.scala

```diff
@@ -34,12 +34,9 @@ private[kafka010] object CachedKafkaProducer extends Logging {
 
   private type Producer = KafkaProducer[Array[Byte], Array[Byte]]
 
-  private val defaultCacheExpireTimeout = TimeUnit.MINUTES.toMillis(10)
-
   private lazy val cacheExpireTimeout: Long =
-    Option(SparkEnv.get).map(_.conf.getTimeAsMs(
-      "spark.kafka.producer.cache.timeout",
-      s"${defaultCacheExpireTimeout}ms")).getOrElse(defaultCacheExpireTimeout)
+    SparkEnv.get.conf.get(PRODUCER_CACHE_TIMEOUT)
```

Comment:
I am pretty sure you have to wrap `SparkEnv.get` with an `Option()` call as before.

See https://github.com/apache/spark/pull/22106#discussion_r210977997